feat: User provisioning & Group sync (#1519)
appsmithorg/appsmith-docs: website/docs/advanced-concepts/user-provisioning-group-sync.mdx

---
description: Guide on how to configure SCIM
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Setup SCIM Provisioning

:::info
SCIM Provisioning is only available in Appsmith's [Enterprise Edition](https://www.appsmith.com/pricing).
:::

> This page shows how to set up SCIM on your Appsmith instance to streamline the user lifecycle management based on your SSO provider.

Reviewer: ```suggestion
This page shows how to configure Appsmith for SCIM to enable the automatic provisioning of users and groups based on your SSO provider.
```

## Prerequisites

- A self-hosted Appsmith instance. See the [installation guides](/getting-started/setup/installation-guides) for deploying Appsmith.

> - A SAML SSO provider configured on your Appsmith instance.

Reviewer: ```suggestion
- A SAML SSO provider configured on your Appsmith instance. See [SAML SSO](/getting-started/setup/instance-configuration/authentication/security-assertion-markup-language-saml) for instructions.
```

## Enable SCIM provisioning on Appsmith

1. On your instance, go to **Admin Settings > Access Control** and click **Provisioning**.
2. On the **User Provisioning & Group Sync** page, click **Configure**.
3. Copy the **SCIM API endpoint** URL to add it later in your IdP application.
4. Under **API key to setup SCIM**, click the **Generate API key** button and copy the API key to add it later in your IdP application.

> 

Reviewer: add screenshot on the setup page on Appsmith

## Configure SSO provider

To complete the setup, go to your IdP application that is connected to Appsmith and follow the steps below:

<Tabs queryString="scim">

> <TabItem label="Active Direcotry (SAML)" value="scim-ad-saml">

Reviewer: ```suggestion
<TabItem label="Active Directory (SAML)" value="scim-ad-saml">
```

Reviewer: I think its misspelt in other places also

> 

Reviewer: ```suggestion
### Configure the enterprise application
```

1. On the Active Directory homepage, click **Enterprise application** on the sidebar and open the SAML application that is already connected to Appsmith.

2. On the application **Overview** page, click **Provisioning** on the sidebar.

3. On the **Provisioning** homepage, click **Provisioning** on the sidebar and follow the steps below:

    a. Select **Automatic** from the **Provision Mode** dropdown.

    b. In the **Tenant URL** field, paste the **SCIM API endpoint** URL that you copied in step 3 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.

    c. In the **Secret Token** field, paste the **API key** that you copied in step 4 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.

    d. Click the **Test Connection** button to check if the added credentials are authorized to enable provisioning.

> 

Reviewer: This section needs to have a separate heading

**Assign users and groups to the application**

    - **userPrincipalName**
    - **displayName**
    - **Switch([IsSoftDeleted], , "False", "True", "True", "False")**

    f. Click **Save**.

    g. Go back to **Mappings** on the **Provisioning** page, click **Provision Azure Active Directory Groups**, and scroll down to **Attribute Mappings** to configure attributes for groups. Delete all the attributes except the following:

    - **displayName**
    - **members**

    h. Click **Save**.

    i. Go back to the **Provisioning** page, turn the toggle **On** for **Provisioning status** present at the bottom. 

4. Go to **Users and Groups** from the sidebar, and click **Add user/group** to add a user or group to your application. These users or groups get automatically synced and added to your Appsmith instance.

5. If you want to sync a user or group manually, go to the **Provision on Demand** from the sidebar, and enter a user or group name in the **Select a user or group** field. 

</TabItem>

<TabItem label="Okta (SAML)" value="scim-okta-saml">

> 

Reviewer: ```suggestion
### Add SCIM provisioning
```

1. On the Okta homepage, go to **Applications > Applications** on the sidebar, and open the SAML application that is already connected to Appsmith.
2. On the application page, go to the **General** tab, click **Edit** on **App Settings** and select the **Enable SCIM provisioning** checkbox. Click **Save**.

> 3. Go to the **Provisioning** tab, click **Edit** on SCIM connection.

Reviewer: Above this add, 

### Choose provisioning options

2. On the application **Overview** page, click **Provisioning** on the sidebar.

3. On the **Provisioning** homepage, click **Provisioning** on the sidebar and follow the steps below:

    a. Select **Automatic** from the **Provision Mode** dropdown.

    b. In the **Tenant URL** field, paste the **SCIM API endpoint** URL that you copied in step 3 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.

    c. In the **Secret Token** field, paste the **API key** that you copied in step 4 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.

    d. Click the **Test Connection** button to check if the added credentials are authorized to enable provisioning.

    e. Expand the **Mappings** section, click **Provision Azure Active Directory Users**, and scroll down to **Attribute Mappings** to configure attributes for users. Delete all the attributes except the following:

    - **userPrincipalName**
    - **displayName**

>     - **Switch([IsSoftDeleted], , "False", "True", "True", "False")**

Author: Ignore the Linter error for this. This is a UI element in AD that needs to be mentioned as it is in the product

> ## Configure Identity provider

Reviewer 2: I in Identity should be lowercase right?
